assets/js/googlesitekit/widgets/datastore/areas.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * core/widgets data store: widgets info.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Store our widget components by registry, then by widget `slug`. We do this because
+ * we can't store React components in our data store.
+ *
+ * @private
+ * @since n.e.x.t
+ */
+export const WidgetComponents = {};
+
+const ASSIGN_WIDGET_AREA = 'ASSIGN_WIDGET_AREA';
+const REGISTER_WIDGET_AREA = 'REGISTER_WIDGET_AREA';
+
+export const INITIAL_STATE = {
+	areas: {},
+	areaWidgets: {},
+	contexts: {
+		analytics: [],
+		dashboard: [],
+	},
+};
+
+export const actions = {
+	/**
+	 * Assigns a widget area to one (or several) contexts.
+	 *
+	 * Accepts an area slug to register as the first argument, then either a string
+	 * (for a single context slug) or array of contexts slugs (to assign the widget area
+	 * to multiple contexts).
+	 *
+	 * Does not error if any of the requested contexts do not exist, but will console.warn
+	 * a message about non-existant contexts.
+	 *
+	 * Does not error if the area slug provided does not exist, but will console.warn a
+	 * message about non-existent areas.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param  {string}         areaSlug     Widget Area's slug.
+	 * @param  {(string|Array)} contextSlugs Widget Context's slug(s).
+	 * @return {Object}                      Redux-style action.
+	 */
+	assignWidgetArea( areaSlug, contextSlugs ) {
+		let contextSlugsAsArray;
+		if ( typeof contextSlugs === 'string' ) {
+			contextSlugsAsArray = [ contextSlugs ];
+		} else {
+			contextSlugsAsArray = [ ...contextSlugsAsArray ];
+		}
```

Comment:
Same comment as above, I don't think we need to introduce a separate variable here?

Reply:
I'm allergic to ternary operators, but it's much nicer here, agreed 👍

Reply:
Curious to hear why? :) I think for simple conditions and assignments it can be easier to read.